integration/commands_administration_test.go

```diff
@@ -986,3 +986,81 @@ func TestCommandsAdministrationCurrentOp(t *testing.T) {
 	_, ok := must.NotFail(doc.Get("inprog")).(*types.Array)
 	assert.True(t, ok)
 }
+
+func TestCommandsAdministrationListIndexes(t *testing.T) {
+	t.Parallel()
+
+	ctx, collection := setup.Setup(t, shareddata.DocumentsStrings)
+
+	for name, tc := range map[string]struct { //nolint:vet // for readability
+		collectionName any
+		expectedResult bson.D
+		expectedError  *mongo.CommandError
+		skip           string
+	}{
+		"existing-collection-default-indexes": {
+			collectionName: collection.Name(),
+			expectedResult: bson.D{
+				{
+					"cursor", bson.D{
+						{"id", int64(0)},
+						{"ns", collection.Database().Name() + "." + collection.Name()},
+						{
+							"firstBatch", bson.A{
+								bson.D{
+									{"v", int32(2)},
+									{"key", bson.D{{"_id", int32(1)}}},
+									{"name", "_id_"},
+								},
+							},
+						},
+					},
+				},
+				{"ok", float64(1)},
+			},
+			expectedError: nil,
+			skip:          "https://github.com/FerretDB/FerretDB/issues/1384",
+		},
+		"non-existent-collection": {
+			collectionName: "non-existent-collection",
+			expectedResult: nil,
+			expectedError: &mongo.CommandError{
+				Code:    26,
+				Name:    "NamespaceNotFound",
+				Message: fmt.Sprintf("ns does not exist: %s.%s", collection.Database().Name(), "non-existent-collection"),
+			},
+		},
+		"invalid-collection-name": {
+			collectionName: 42,
+			expectedResult: nil,
+			expectedError: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "collection name has invalid type int",
+			},
+		},
+	} {
+		name, tc := name, tc
+
+		t.Run(name, func(t *testing.T) {
+			if tc.skip != "" {
+				t.Skip(tc.skip)
+			}
+
+			var res bson.D
+			err := collection.Database().RunCommand(
```

Comment:
Let's test `collection.Indexes()`

Reply:
That produces a data race on the driver side 😅 
I pushed the code that demonstrates it.

Reply:
Added `Indexes().List()` for a compat test.